esmf_regrid/tests/unit/esmf_regridder/test_Regridder.py

```diff
@@ -86,35 +85,111 @@ def test_Regridder_regrid():
 
     # Regrid with a fully masked array.
     result_fullmask = rg.regrid(ma.array(src_array, mask=True))
-    expected_fulmask = ma.array(np.zeros([3, 2]), mask=True)
+    expected_fulmask = ma.array(np.zeros([2, 3]), mask=True)
     assert ma.allclose(result_fullmask, expected_fulmask)
 
     # Regrid with a masked array containing a masked point.
     result_withmask = rg.regrid(src_masked)
     expected_withmask = ma.array(
         [
-            [0.9999999999999999, 1.0],
-            [0.7503444126612077, 0.4999999999999997],
-            [0.6674194025656824, 0.0],
+            [0.9999999999999999, 0.7503444126612077, 0.6674194025656824],
+            [1.0, 0.4999999999999997, 0.0],
         ]
     )
     assert ma.allclose(result_withmask, expected_withmask)
 
     # Regrid while setting mdtol.
     result_half_mdtol = rg.regrid(src_masked, mdtol=0.5)
-    expected_half_mdtol = ma.array(expected_withmask, mask=[[1, 0], [0, 0], [1, 0]])
+    expected_half_mdtol = ma.array(
+        expected_withmask, mask=np.array([[1, 0, 1], [0, 0, 0]])
+    )
     assert ma.allclose(result_half_mdtol, expected_half_mdtol)
 
     # Regrid with norm_type="dstarea".
     result_dstarea = rg.regrid(src_masked, norm_type="dstarea")
     expected_dstarea = ma.array(
         [
-            [0.3325805974343169, 0.9999999999999998],
-            [0.4999999999999999, 0.499931367542066],
-            [0.6674194025656823, 0.0],
+            [0.3325805974343169, 0.4999999999999999, 0.6674194025656823],
+            [0.9999999999999998, 0.499931367542066, 0.0],
         ]
     )
     assert ma.allclose(result_dstarea, expected_dstarea)
 
+    assert src_array.T.shape != src_array.shape
+    with pytest.raises(ValueError):
+        _ = rg.regrid(src_array.T)
+
     with pytest.raises(ValueError):
         _ = rg.regrid(src_masked, norm_type="INVALID")
+
+
+def test_Regridder_init_small():
+    """
+    Simplified test for :meth:`~esmf_regrid.esmf_regridder.Regridder.regrid`.
+
+    This test is designed to be simple enough to generate predictable weights.
+    With predictable weights it is easier to check that the weights are
+    associated with the correct indices.
+    """
+    # The following ASCII visualisation describes the source and target grids
+    # and the indices which ESMF assigns to their cells.
+    # Analysis of the weights dict returned by ESMF confirms that this is
+    # indeed the indexing which ESMF assigns these grids.
+    #
+    # 30  +---+---+       +-------+
+    #     | 3 | 6 |       |       |
+    # 20  +---+---+       |   2   |
+    #     | 2 | 5 |       |       |
+    # 10  +---+---+       +-------+
+    #     | 1 | 4 |       |   1   |
+    #  0  +---+---+       +-------+
```

Comment:
The tgt cell 1, also overlaps with a tiny area of src cells 2 and 5. (this confused me for a while)

Reply:
This is addressed in a comment below.